packages/core/src/components/axes/grid-brush.ts

```diff
@@ -97,87 +103,76 @@ export class ChartBrush extends Component {
-						event.sourceEvent.type === "mousedown" ||
-						event.sourceEvent.type === "touchstart" ||
-						event.sourceEvent.type === "touchmove" ||
-						event.sourceEvent.type === "touchend")
+					zoomDomain[0].valueOf() !== newDomain[0].valueOf() ||
+					zoomDomain[1].valueOf() !== newDomain[1].valueOf()
 				) {
-					// dispatch selection events
-					let zoomBarEventType;
-					if (event.type === "start") {
-						zoomBarEventType = Events.ZoomBar.SELECTION_START;
-					} else if (event.type === "brush") {
-						zoomBarEventType = Events.ZoomBar.SELECTION_IN_PROGRESS;
-					} else if (event.type === "end") {
-						zoomBarEventType = Events.ZoomBar.SELECTION_END;
-					}
-					this.services.events.dispatchEvent(zoomBarEventType, {
-						selection,
-						newDomain
-					});
+					this.services.zoom.handleDomainChange(newDomain);
 				}
 			};
+
 			const brushed = () => {
+				// max selection range: [0, width]
 				const selection = event.selection;
 
 				if (selection !== null) {
-					// create xScale based on current zoomDomain
-					const xScale = scaleTime()
-						.range([0, width])
-						.domain(zoomDomain);
-
-					let newDomain = [
-						xScale.invert(selection[0]),
-						xScale.invert(selection[1])
-					];
-
-					// if selected start time and end time are the same
-					// reset to default full range
-					if (newDomain[0].valueOf() === newDomain[1].valueOf()) {
-						// same as d3 behavior and zoom bar behavior: set to default full range
-						newDomain = this.services.zoom.getDefaultZoomBarDomain();
-					}
-
-					// only if zoomDomain needs update
-					if (
-						zoomDomain[0].valueOf() !== newDomain[0].valueOf() ||
-						zoomDomain[1].valueOf() !== newDomain[1].valueOf()
-					) {
-						this.model.set(
-							{ zoomDomain: newDomain },
-							{ animate: false }
-						);
-					}
+					// handleZoomDomain assumes max range is [0, width]
+					handleZoomDomain(selection[0], selection[1]);
 
 					// clear brush selection
 					brushArea.call(brush.move, null);
 					// hide frontSelection
 					frontSelection.style("display", "none");
 				}
 			};
-
 			// leave some space to display selection strokes besides axis
 			const brush = brushX()
 				.extent([
 					[0, 0],
 					[width - 1, height]
 				])
-				.on("start brush end", eventHandler)
+				.on("start brush end", brushEventHandler)
 				.on("end.brushed", brushed);
 
 			brushArea.call(brush);
+
+			const zoomRatio = this.services.zoom.getZoomRatio();
+			backdrop.on("click", function () {
+				if (event.shiftKey) {
+					// clickedX range: [0, width]
+					const clickedX = mouse(brushArea.node())[0];
+
+					let leftPoint = clickedX - (width * zoomRatio) / 2;
+					if (leftPoint < 0) {
+						leftPoint = 0;
+					}
+					let rightPoint = clickedX + (width * zoomRatio) / 2;
+					if (rightPoint > width) {
+						rightPoint = width;
+					}
+					// handleZoomDomain assumes max range is [0, width]
+					handleZoomDomain(leftPoint, rightPoint);
```

Comment:
`handleZoomDomain` might not be the best name, what does this function do? It confuses me a bit too

Reply:
looks like it's updating the zoom domain, maybe `updateZoomDomain`?

Reply:
I am ok with `updateZoomDomain`.  Updated in https://github.com/carbon-design-system/carbon-charts/pull/773/commits/fb1d4ab541e920409606016791cc45cbe2872896.